### PR TITLE
Adding javadoc to explain that items will be continually be emitted u…

### LINKED ISSUE #2135: Update `asObservable` JavaDoc to mention that they keep emitting and never call onComplete
I request only once to my method that uses realm query with asObservable.
(I saw Log for sure that I requested really once, and it's true.)

But subscribe event from the query which is observable came multiple, twice sometimes triple.

Besides, one of the subscribe events gives me wrong results as 0, even if table has some datas.

How can I get subscribe event only once when I request once query with observable?

### PR DESCRIPTION
**Pull request by donnfelker**
…nless other wise instructed with the first() operator.

Closes #2135

### CODE REVIEW

File: realm/realm-library/src/main/java/io/realm/BaseRealm.java [hunk 1]
```diff
@@ -167,8 +167,21 @@ public void removeChangeListener(RealmChangeListener listener) {
     }
 
     /**
-     * Returns an Rx Observable that monitors changes to this Realm. It will emit the current state when subscribed
-     * to.
+     * Returns an Rx Observable that monitors changes to this Realm. It will emit the current state
+     * when subscribed to. Items will continually be emitted as the Realm is updated -
+     * {@code onComplete} will never be called.
+     *
+     * If you would like the {@code asObservable()} to stop emitting items you can instruct Rx to
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/BaseRealm.java]: Is it RxJava or just Rx ?

File: realm/realm-library/src/main/java/io/realm/BaseRealm.java [hunk 2]
```diff
@@ -167,8 +167,21 @@ public void removeChangeListener(RealmChangeListener listener) {
...
+     * only emit only the first item by using the {@code first()} operator:
+     *
+     * <pre>
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/BaseRealm.java]: Do you need both `<pre>` and `{@code` I thought it was the same thing?

**Comment by cmelchior** [comment]
I like it, and great with the example as well. We need to update the other `asObservable` methods as well though:  `RealmResults`, `RealmObject`

File: realm/realm-library/src/main/java/io/realm/BaseRealm.java (hunk 1 above)
Comment (donnfelker) [inline realm/realm-library/src/main/java/io/realm/BaseRealm.java]: I should make it RxJava. Will fix.

File: realm/realm-library/src/main/java/io/realm/BaseRealm.java (hunk 2 above)
Comment (donnfelker) [inline realm/realm-library/src/main/java/io/realm/BaseRealm.java]: Which do you prefer to use?

**Comment by donnfelker** [comment]
Ahh, will do. I'll update the other classes.

**Comment by donnfelker** [comment]
@cmelchior Updated the RealmObject and RealmResults. Let me know about the code formatting <pre> and {@code} preference and I'll update it.

**Comment by cmelchior** [comment]
Looks like I was wrong, It seems that using both `<pre>` and `{@code` is correct: http://stackoverflow.com/questions/541920/multiple-line-code-example-in-javadoc-comment.

File: realm/realm-library/src/main/java/io/realm/BaseRealm.java [hunk 3]
```diff
@@ -167,8 +167,21 @@ public void removeChangeListener(RealmChangeListener listener) {
...
+     * Returns an RxJava Observable that monitors changes to this Realm. It will emit the current state
+     * when subscribed to. Items will continually be emitted as the Realm is updated -
+     * {@code onComplete} will never be called.
+     *
+     * If you would like the {@code asObservable()} to stop emitting items you can instruct RxJava to
+     * only emit only the first item by using the {@code first()} operator:
+     *
+     * <pre>
+     * {@code
+     * realm.where(Foo.class).findAllAsync().asObservable()
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/BaseRealm.java]: I would adjust the example to fit the context. In this one, we are observing on a Realm:

```
realm.asObservable().first()
```

File: realm/realm-library/src/main/java/io/realm/RealmObject.java [hunk 4]
```diff
@@ -264,12 +264,25 @@ public final void removeChangeListeners() {
...
      * If chaining a RealmObject observable use {@code obj.<MyRealmObjectClass>asObservable()} to pass on
      * type information, otherwise the type of the following observables will be {@code RealmObject}.
      *
+     * If you would like the {@code asObservable()} to stop emitting items you can instruct RxJava to
+     * only emit only the first item by using the {@code first()} operator:
+     *
+     * <pre>
+     * {@code
+     * obj.asObservable()
+     *      .filter(obj->isLoaded())
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/RealmObject.java]: Isn't the proper syntax `obj::isLoaded` or `obj -> obj.isLoaded()` ?

File: realm/realm-library/src/main/java/io/realm/RealmResults.java [hunk 5]
```diff
@@ -900,7 +900,20 @@ public void removeChangeListeners() {
...
-     * subscribed to.
+     * subscribed to. RealmResults will continually be emitted as the RealmResults are updated -
+     * {@code onComplete} will never be called.
+     *
+     * If you would like the {@code asObservable()} to stop emitting items you can instruct RxJava to
+     * only emit only the first item by using the {@code first()} operator:
+     *
+     * <pre>
+     * {@code
+     * results.asObservable()
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/RealmResults.java]: I would use the async example from BaseRealm here instead.

File: realm/realm-library/src/main/java/io/realm/RealmResults.java [hunk 6]
```diff
@@ -900,7 +900,20 @@ public void removeChangeListeners() {
...
+     *      .filter(results->isLoaded())
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/RealmResults.java]: Same, correct syntax?

File: realm/realm-library/src/main/java/io/realm/RealmObject.java (hunk 4 above)
Comment (donnfelker) [inline realm/realm-library/src/main/java/io/realm/RealmObject.java]: You're right, it was the second one. :)

**Comment by donnfelker** [comment]
@cmelchior All requested changes have been made. :)

**Comment by cmelchior** [comment]
Looks good to me. :+1: once squashed.
@realm/java

**Comment by beeender** [comment]
:+1:

**Comment by donnfelker** [comment]
@cmelchior Squashed.

**Comment by cmelchior** [comment]
Hmm, looks like a regression on our part somewhere: io.realm.RealmSchemaTests.remove_classWithReferencesThrows (from io.realm.DynamicRealmObjectTests)

**Comment by cmelchior** [comment]
Retest this please

**Comment by donnfelker** [comment]
Do you need me to do anything to help push this along?

**Comment by cmelchior** [comment]
No, this PR shouldn't cause any build failure. I'll look into it.

**Pull request merged**